src/main/java/edu/harvard/iq/dataverse/api/Admin.java

```diff
@@ -1920,4 +1926,56 @@ public Response getBannerMessages(@PathParam("id") Long id) throws WrappedRespon
 
     }
     
+    @GET
+    @Path("/licenses")
+    public Response getLicenses() {
+        return ok(licenseService.listAll().stream()
+        		.map(JsonPrinter::json)
+        		.collect(toJsonArray()));
+    }
+
+    @GET
+    @Path("/licenses/{id}")
+    public Response getLicense(@PathParam("id") long id) {
+    	License l = licenseService.get(id);
+    	if (l == null) {
+    		return error(Response.Status.NOT_FOUND, "Not Found.");
+		}
+		return ok(json(l));
+    }
+
+    @POST
+    @Path("/licenses")
+    public Response addLicense(License l) {
+		try {
+			License added = licenseService.save(l);
+			if (added == null) {
+				return error(Response.Status.BAD_REQUEST, "Bad Request.");
```

Comment:
Do we know for certain that Bad Request is appropriate here? If you throw specific exceptions for specific conditions and catch them here you can ensure that the client gets an informative error message (and at a minimum a non-misleading one).